app/models/campaign_page.rb

```diff
@@ -27,4 +30,9 @@ class CampaignPage < ActiveRecord::Base
   def create_slug
     self.slug = title.parameterize if slug.nil? and not title.nil?
   end
+
+  # Compiles the HTML for this CampaignPage so that it can be used by external display apps.
+  def compile_html
+    self.compiled_html = render partial: 'campaign_pages/page_compile', layout: false, locals: {campaign_page: self}
```

Comment:
I'm not comfortable having a model compile its own HTML - it's a fat violation of SRP.

Really there should be a dedicated class for this:

```ruby
class CampaignPageRender
  include RenderAnywhere

  def initialize(page)
    @page = page
  end

  def render_and_save
    @page.update_attributes(compiled_html: html)
  end 

  private

  def html
    render partial: 'campaign_pages/page_compile', layout: false, locals: {campaign_page: @page}
  end
end
```